### PR TITLE
BREAKING CHANGE(core/caniuse): new browser compat header

### LINKED ISSUE #4042: Consider displaying the first version a feature started to work at in CanIUse widget
The current widget shows the _latest_ version an API can be used at:

Screen Shot 2022-02-17 at 11 17 01

([Source](https://www.w3.org/TR/geolocation/))

It might be more useful to show the _first_ version an API started to work in. This is a widget based on the same data our team has built for [web.dev](https://web.dev/):

Screen Shot 2022-02-17 at 11 16 27

([Source](https://web.dev/file-system-access/#browser-support))

CC: @marcoscaceres

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #4042

With its breathtaking new design, the all new caniuse header is now 30% more flexy, conveying more information in fraction of the space. It's the best caniuse header we've ever built. 
 
Screen Shot 2022-03-10 at 8 01 44 pm

### CODE REVIEW

File: src/core/caniuse.js [hunk 1]
```diff
@@ -98,48 +101,85 @@ export async function run(conf) {
   }
 }
 
+function handleError(err, options, featureURL) {
+  const msg = `Couldn't find feature "${options.feature}".`;
```
Comment (sidvishnoi) [inline src/core/caniuse.js:105]: This is assuming 404. It could be 50x too.

**Review by sidvishnoi — CHANGES_REQUESTED** [review]

File: src/core/caniuse.js (hunk 1 above)
Comment (marcoscaceres) [inline src/core/caniuse.js:105]:
```suggestion
  const msg = `Failed to retrieve feature "${options.feature}".`;
```

File: src/core/caniuse.js [hunk 2]
```diff
@@ -98,48 +101,85 @@ export async function run(conf) {
...
+async function processJson(json, { feature }) {
+  const results = json.result;
+  const out = results.map(({ browser: browserId, version, caniuse }) => {
+    const { name } = BROWSERS.get(browserId);
+    const versionLong = version ? ` version ${version}` : "";
+    const browserName = `${name}${versionLong}`;
+    const supportLevel = statToText.get(caniuse);
+    const ariaLabel = `${feature} is ${supportLevel} in ${browserName}.`;
+    const cssClass = `caniuse-cell ${caniuse}`;
+    const title = capitalize(`${supportLevel} in ${browserName}.`);
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply src/core/caniuse.js:144]: Sadly, that makes the capitalization of the `ariaLabel` wrong.

File: tests/spec/core/caniuse-spec.js [hunk 3]
```diff
@@ -78,35 +71,44 @@ describe("Core — Can I Use", () => {
...
-    const moreInfoLink = stats.querySelector("a");
+    const cells = stats.querySelectorAll(".caniuse-cell");
+    expect(cells).toHaveSize(4);
+
+    // Check a cell
+    const [cell] = cells;
+    expect(cell.title).toBe(
+      "Supported by default in Android Chrome version 78."
+    );
+    expect(cell.getAttribute("area-label")).toBe(
```
Comment (marcoscaceres) [inline tests/spec/core/caniuse-spec.js:87]:
```suggestion
    expect(cell.getAttribute("aria-label")).toBe(
```

**Comment by marcoscaceres** [comment]
Taking as is, as I need to record a video that uses it.

File: src/core/caniuse.js [hunk 4]
```diff
@@ -98,48 +101,85 @@ export async function run(conf) {
...
```
Comment (sidvishnoi) [inline src/core/caniuse.js:144]: Should use "since" instead of "in" I think.

File: src/core/caniuse.js [hunk 5]
```diff
@@ -98,48 +101,85 @@ export async function run(conf) {
...
+}
+
+async function processJson(json, { feature }) {
+  const results = json.result;
+  const out = results.map(({ browser: browserId, version, caniuse }) => {
+    const { name } = BROWSERS.get(browserId);
+    const versionLong = version ? ` version ${version}` : "";
+    const browserName = `${name}${versionLong}`;
+    const supportLevel = statToText.get(caniuse);
+    const ariaLabel = `${feature} is ${supportLevel} in ${browserName}.`;
```
Comment (sidvishnoi) [inline src/core/caniuse.js:142]: Firefox a11y panel shows it like this:
image

Should test it out with Voiceover/NVDA

File: src/core/caniuse.js [hunk 6]
```diff
@@ -98,48 +101,85 @@ export async function run(conf) {
...
-  searchParams.set("feature", feature);
-  searchParams.set("versions", versions);
-  if (Array.isArray(browsers)) {
-    searchParams.set("browsers", browsers.join(","));
-  }
-  searchParams.set("format", "html");
-  const url = `${apiURL}?${searchParams.toString()}`;
+async function fetchStats(options) {
+  const { feature, browsers, apiURL } = options;
+  const url = new URL(apiURL || `./${feature}`, API_URL);
```
Comment (sidvishnoi) [inline src/core/caniuse.js:173]: I think we wanted:
```suggestion
  const url = new URL(`./${feature}`, apiURL || API_URL);
```

**Pull request merged**